src/Http/Role/EditRoleRestController.php

```diff
@@ -233,4 +233,13 @@ private function getLabelId(string $labelIdentifier): ?UUID
             return is_null($entity) ? null : $entity->getUuid();
         }
     }
+
+    /**
+     * This function was added during the refactoring of Permission.
+     * Because the API expects Uppercase/Underscore names for Permission.
+     */
+    public static function getByUppercaseName(string $key): Permission
+    {
+        return new Permission(ucfirst(str_replace('_', ' ', strtolower($key))));
+    }
```

Comment:
Wouldn't it be easier and less error-prone to keep the old uppercased names with underscores instead? Now we have to make sure we don't overlook a place where the conversion needs to happen. And it feels confusing in the long run. Someone might accidentally use the new casing in new endpoints which would be inconsistent. Was there a specific reason to change the casing of these permission values?

Reply:
But the fact that this is copied in various classes again makes me wonder if this is the ideal approach. Maybe it should at least be copied to `Permission` itself if we keep it.

Reply:
We wanted to create a follow-up ticket to refactor the usage of the uppercase values. Because only the UI needs the uppercase values inside the admin interface to manage roles. That was also the reason to keep the method `getUppercaseName` away from the `Permission` class.

Reply:
Assuming that only the UI uses these uppercase values for a second (not 100% sure), it will still require changes in a lot of UI repositories:

- udb3-frontend
- udb3-angular
- udb3-angular-app
- omd-angular-app

So like I asked in my first comment, was there a specific reason to change the casing of these permission values?

Technically it's a breaking API change if we change the casing. Best case we have to update 4 repositories, but I wouldn't be surprised if some other internal tool also uses them.

Reply:
> Because only the UI needs the uppercase values inside the admin interface to manage roles.

The UI also uses it to actually check permissions, like which links to show in the sidebar.

Reply:
When discussing this PR with Jonas I first made the advise to make the values lowercase because those lowercase values are stored inside the database. But I think I gave some bad advise. It seems better to keep the values as uppercase values. Inside the database layer we have more control on how things are handled compared to the parts that consume the exposed API.

One question stays, if we make the values uppercase, do we also make the method names uppercase? I guess we could keep the method names lowercase because that matches the method naming convention. But the values they return are uppercase.

Reply:
I think the current approach is fine:
- The `Permission` class exposes lower case values
- The database and business layer work with the normal lower case values
- The API layer uses well documented methods that do a upper case conversion